zipline-kotlin-plugin-hosted/src/main/kotlin/app/cash/zipline/kotlin/AdapterGenerator.kt

```diff
@@ -763,16 +765,32 @@ internal class AdapterGenerator(
       context = pluginContext,
       scopeOwnerSymbol = result.symbol
     ) {
-      val newCall = irCall(ziplineApis.outboundBridgeContextNewCall).apply {
-        dispatchReceiver = irCall(
+      val contextLocal = irTemporary(
+        value = irCall(
           callee = outboundContextProperty.getter!!
         ).apply {
           dispatchReceiver = irGet(result.dispatchReceiverParameter!!)
-        }
+        },
+        nameHint = "context",
+        isMutable = false
+      ).apply {
+        origin = IrDeclarationOrigin.DEFINED
+      }
+
+      val newCall = irCall(ziplineApis.outboundBridgeContextNewCall).apply {
+        dispatchReceiver = irGet(contextLocal)
         putValueArgument(0, irString(bridgedFunction.signature))
         putValueArgument(1, irInt(bridgedFunction.valueParameters.size))
       }
 
+      // If this is the close() function, tell the OutboundContext that this instance is closed.
```

Comment:
Can you add a comment that matches what the source code would do?
```
// context.closed = true
```
I find it really helpful in IR stuff.

Reply:
done